### ISSUE TITLE
[HCL2] Vagrant post-processor: keep_input_artifact does not work

### ISSUE THREAD

**Issue by minhluantran017** [hashicorp/packer#9460]
#### Overview of the Issue

[HCL2] Vagrant post-processor: keep_input_artifact does not work

#### Reproduction Steps
Vagrant post-processor in Packer HCL template:
```hcl
    post-processor "vagrant" {
        name = "vagrant-box"
        output = "output-${var.build_number}/packer-ubuntu1404-${var.build_number}.box"
        keep_input_artifact = true
    }
```
The `keep_input_artifact` does not work as expected:
```
ubuntu@Thinkpad-E490s:~/demo-packer/packer-virtualbox-iso$ packer validate ubuntu1404.pkr.hcl 
Error: Unsupported argument

  on ubuntu1404.pkr.hcl line 95:
  (source code not available)

An argument named "keep_input_artifact" is not expected here.


ubuntu@Thinkpad-E490s:~/demo-packer/packer-virtualbox-iso$
```
### Packer version
```
ubuntu@Thinkpad-E490s:~/demo-packer/packer-virtualbox-iso$ packer version
Packer v1.6.0
ubuntu@Thinkpad-E490s:~/demo-packer/packer-virtualbox-iso$ 
```

**Comment by SwampDragons** [comment]
Can you please share your entire template so we can reproduce?

**Comment by azr** [comment]
Hello there, thanks for opening; the `keep_input_artifact` field was not made available for HCL2 users; #9477 solves this !
